### ISSUE TITLE
No longer creates dest directory

### ISSUE THREAD

**Issue by gnomesoftuk** [mlison/protractor-jasmine2-screenshot-reporter#43]
Having upgraded to 0.30 my dest directory is no longer being created automatically. The first time the reporter attempts to access it fails. If I create the dest directory manually the reporter works fine.

Protractor Config:

jasmine.getEnv().addReporter(
            new HtmlScreenshotReporter({
                dest: 'target/screenshots/',
                cleanDestination: true,
                filename: 'E2E-test-report.html',
                ignoreSkippedSpecs: true,
                captureOnlyFailedSpecs: true,
                showSummary: true
            })
        );

Npm config :

"protractor": "^3.0.0",
"protractor-jasmine2-screenshot-reporter": "^0.3.0",
"jasmine-core": "^2.4.1",
"jasmine-reporters": "^2.1.1",
"selenium-webdriver": "^2.48.2",
"webdriver-manager": "^8.0.0"

If I revert to 0.20 and do nothing else, the problem goes away.

Error :
[firefox #2] .Error: ENOENT: no such file or directory, open 'target/screenshots/E2E-test-report.html'
[firefox #2]     at Error (native)
[firefox #2]     at Object.fs.openSync (fs.js:549:18)
[firefox #2]     at Object.fs.writeFileSync (fs.js:1155:15)
[firefox #2]     at Object.fs.appendFileSync (fs.js:1203:6)
[firefox #2]     at Jasmine2ScreenShotReporter.jasmineDone (/mnt/storage/dev_workspace_performance/workspace/Projects/shoal-platform/shoal-client/node_modules/protractor-jasmine2-screenshot-reporter/index.js:482:10)
[firefox #2]     at dispatch (/mnt/storage/dev_workspace_performance/workspace/Projects/shoal-platform/shoal-client/node_modules/jasmine/node_modules/jasmine-core/lib/jasmine-core/jasmine.js:1869:28)
[firefox #2]     at /mnt/storage/dev_workspace_performance/workspace/Projects/shoal-platform/shoal-client/node_modules/jasmine/node_modules/jasmine-core/lib/jasmine-core/jasmine.js:1852:11
[firefox #2]     at QueueRunner.clearStack (/mnt/storage/dev_workspace_performance/workspace/Projects/shoal-platform/shoal-client/node_modules/jasmine/node_modules/jasmine-core/lib/jasmine-core/jasmine.js:605:9)
[firefox #2]     at QueueRunner.run (/mnt/storage/dev_workspace_performance/workspace/Projects/shoal-platform/shoal-client/node_modules/jasmine/node_modules/jasmine-core/lib/jasmine-core/jasmine.js:1784:12)
[firefox #2]     at /mnt/storage/dev_workspace_performance/workspace/Projects/shoal-platform/shoal-client/node_modules/jasmine/node_modules/jasmine-core/lib/jasmine-core/jasmine.js:1801:16

**Comment by AndrewFinlay** [comment]
It sounds like you'll need to update your protractor config file.  0.3 introduces some big changes to your configuration file to allow for running concurrent tests in sharded mode.  It also includes a few bug fixes aimed at preventing 0.2 from going haywire while running concurrent tests.  Let me know if this fixes the issue.

**Comment by gnomesoftuk** [comment]
Thank you Andrew, I didn't realise the config had changed in this way. Adding the beforeLaunch and afterLaunch sections fixed the issue.

New Config :

var HtmlScreenshotReporter = require('protractor-jasmine2-screenshot-reporter'),

...

```
screenshotReporter = new HtmlScreenshotReporter({
    dest: 'target/screenshots/',
    filename: 'E2E-test-report.html',
    ignoreSkippedSpecs: true,
    captureOnlyFailedSpecs: false,
    showSummary: true
});
```

...

```
beforeLaunch: function () {
    return new Promise(function (resolve) {
        screenshotReporter.beforeLaunch(resolve);
    });
},


onPrepare: function () {
        /** create selenium screenshots **/
        jasmine.getEnv().addReporter(screenshotReporter);
    });
},

afterLaunch: function (exitCode) {
    return new Promise(function (resolve) {
        screenshotReporter.afterLaunch(resolve.bind(this, exitCode));
    });
}
```
